core/trino-main/src/main/java/io/trino/execution/querystats/DefaultQueryStatsCollector.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution.querystats;
+
+import io.trino.sql.planner.iterative.Rule;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+public class DefaultQueryStatsCollector
```

Comment:
why do we need multiple implementations?

Reply:
Similar design as with `WarningCollector` and its `NOOP` implementation.

Reply:
When NOOP is used?

Reply:
Mostly in tests, but there are two usages in LocalQueryRunner.